index.js

```diff
@@ -55,6 +57,7 @@ function ensureAllCommentsPrinted(astComments) {
 }
 
 function formatWithCursor(text, opts, addAlignmentSize) {
+  text = prependFormatIfNecessary(text);
```

Comment:
If you prepend the docblock here, the `stripBom` on the next line won't get a chance to ever strip the BOM character that is now not at the start of the file, but somewhere in the middle.

I think it would be better to prepend the docblock comment _after_ formatting, not before.

If the `rangeStart` and `rangeEnd` options are specified, we shouldn't prepend the docblock, because in this case, we are formatting only a selected range inside the file. The docblock comment will be added in the middle of the file.

There's also the `cursorOffset` option -- it's worth ensuring that it will take the added docblock into account when computing the new offset after formatting.